trilby-cli/src/Trilby/Update.hs

```diff
@@ -3,16 +3,76 @@
 module Trilby.Update (update) where
 
 import Trilby.Update.Options
+import Turtle (readlink)
+import Data.Text qualified as Text
 import Prelude
+import Data.List.Extra (split)
 
 update :: UpdateOpts Maybe -> App ()
-update (askOpts -> opts) = inDir "/etc/trilby" do
+update (askOpts -> opts) = do
     whenM opts.flakeUpdate $ rawCmd_ ["nix", "flake", "update", "--accept-flake-config"]
-    withTrace rawCmd_ ["nixos-rebuild", "build", "--flake", ".", "--accept-flake-config"]
-    rawCmd_ ["nvd", "diff", "/run/current-system", "result"]
+    hosts <- fromListSafe Localhost <$> (fromText <$$> opts.hosts)
+    for_ @NonEmpty hosts $ updateHost opts
+
+updateHost :: UpdateOpts App -> Host -> App ()
+updateHost opts host = inDir "/etc/trilby" do
+    perform NixosRebuildBuild
+    resultPath <- readlink "./result"
+    exec rawCmd_ ["nvd", "diff", "/run/current-system", fromString resultPath]
+    case host of
+        Localhost -> pure ()
+        Host{} -> $(logWarn) $ "Choosing action for host " <> ishow host
     opts.action >>= \case
-        Switch -> (withTrace . asRoot) rawCmd_ ["nixos-rebuild", "switch", "--flake", ".", "--accept-flake-config"]
+        Switch -> perform NixosRebuildSwitch
         Boot{..} -> do
-            (withTrace . asRoot) rawCmd_ ["nixos-rebuild", "boot", "--flake", ".", "--accept-flake-config", "--install-bootloader"]
-            whenM reboot $ cmd_ ["systemctl", "reboot"]
+            perform NixosRebuildBoot
+            whenM reboot $ exec cmd_ ["systemctl", "reboot"]
         NoAction -> pure ()
+    where
+        perform action = nixosRebuild $ NixosRebuild { flakeRef = ".", .. }
+        exec = ssh host
+
+data NixosRebuildAction
+    = NixosRebuildBuild
+    | NixosRebuildSwitch
+    | NixosRebuildBoot
+
+data Host
+    = Host { username :: Maybe Text, hostname :: Text}
+    | Localhost
+
+instance IsString Host where
+    fromString s =
+        case split (== '@') s of
+                ((fromString -> Just -> username):  (fromString -> hostname): _) -> Host{..}
```

Comment:
Nitpick: can we have symmetric `:`s with spaces on both sides? :clown_face:

Reply:
Is there a Fourmolu incantation that will do this?